src/openrct2/object/RideObject.cpp

```diff
@@ -983,99 +983,16 @@ bool RideObject::IsRideTypeShopOrFacility(uint8_t rideType)
-    { "lift", RIDE_TYPE_LIFT },
-    { "vertical_drop_rc", RIDE_TYPE_VERTICAL_DROP_ROLLER_COASTER },
-    { "cash_machine", RIDE_TYPE_CASH_MACHINE },
-    { "twist", RIDE_TYPE_TWIST },
-    { "haunted_house", RIDE_TYPE_HAUNTED_HOUSE },
-    { "first_aid", RIDE_TYPE_FIRST_AID },
-    { "circus", RIDE_TYPE_CIRCUS },
-    { "ghost_train", RIDE_TYPE_GHOST_TRAIN },
-    { "twister_rc", RIDE_TYPE_TWISTER_ROLLER_COASTER },
-    { "wooden_rc", RIDE_TYPE_WOODEN_ROLLER_COASTER },
-    { "side_friction_rc", RIDE_TYPE_SIDE_FRICTION_ROLLER_COASTER },
-    { "steel_wild_mouse", RIDE_TYPE_STEEL_WILD_MOUSE },
-    { "multi_dimension_rc", RIDE_TYPE_MULTI_DIMENSION_ROLLER_COASTER },
-    { "flying_rc", RIDE_TYPE_FLYING_ROLLER_COASTER },
-    { "virginia_reel", RIDE_TYPE_VIRGINIA_REEL },
-    { "splash_boats", RIDE_TYPE_SPLASH_BOATS },
-    { "mini_helicopters", RIDE_TYPE_MINI_HELICOPTERS },
-    { "lay_down_rc", RIDE_TYPE_LAY_DOWN_ROLLER_COASTER },
-    { "suspended_monorail", RIDE_TYPE_SUSPENDED_MONORAIL },
-    { "reverser_rc", RIDE_TYPE_REVERSER_ROLLER_COASTER },
-    { "heartline_twister_rc", RIDE_TYPE_HEARTLINE_TWISTER_COASTER },
-    { "mini_golf", RIDE_TYPE_MINI_GOLF },
-    { "giga_rc", RIDE_TYPE_GIGA_COASTER },
-    { "roto_drop", RIDE_TYPE_ROTO_DROP },
-    { "flying_saucers", RIDE_TYPE_FLYING_SAUCERS },
-    { "crooked_house", RIDE_TYPE_CROOKED_HOUSE },
-    { "monorail_cycles", RIDE_TYPE_MONORAIL_CYCLES },
-    { "compact_inverted_rc", RIDE_TYPE_COMPACT_INVERTED_COASTER },
-    { "water_coaster", RIDE_TYPE_WATER_COASTER },
-    { "air_powered_vertical_rc", RIDE_TYPE_AIR_POWERED_VERTICAL_COASTER },
-    { "inverted_hairpin_rc", RIDE_TYPE_INVERTED_HAIRPIN_COASTER },
-    { "magic_carpet", RIDE_TYPE_MAGIC_CARPET },
-    { "submarine_ride", RIDE_TYPE_SUBMARINE_RIDE },
-    { "river_rafts", RIDE_TYPE_RIVER_RAFTS },
-    { "enterprise", RIDE_TYPE_ENTERPRISE },
-    { "inverted_impulse_rc", RIDE_TYPE_INVERTED_IMPULSE_COASTER },
-    { "mini_rc", RIDE_TYPE_MINI_ROLLER_COASTER },
-    { "mine_ride", RIDE_TYPE_MINE_RIDE },
-    { "lim_launched_rc", RIDE_TYPE_LIM_LAUNCHED_ROLLER_COASTER },
-    { "hypercoaster", RIDE_TYPE_HYPERCOASTER },
-    { "hyper_twister", RIDE_TYPE_HYPER_TWISTER },
-    { "monster_trucks", RIDE_TYPE_MONSTER_TRUCKS },
-    { "spinning_wild_mouse", RIDE_TYPE_SPINNING_WILD_MOUSE },
-    { "classic_mini_rc", RIDE_TYPE_CLASSIC_MINI_ROLLER_COASTER },
-    { "hybrid_rc", RIDE_TYPE_HYBRID_COASTER },
-    { "single_rail_rc", RIDE_TYPE_SINGLE_RAIL_ROLLER_COASTER },
-};
-
 uint8_t RideObject::ParseRideType(const std::string& s)
 {
-    auto result = RideTypeLookupTable.find(s);
-    return (result != RideTypeLookupTable.end()) ? result->second : static_cast<uint8_t>(RIDE_TYPE_NULL);
+    uint8_t rideType;
+    for (rideType = 0; rideType < RIDE_TYPE_COUNT; rideType++)
+    {
+        const auto& rtd = GetRideTypeDescriptor(rideType);
+        if (rtd.Name == s)
+            return rideType;
+    }
+    return RIDE_TYPE_NULL;
```

Comment:
The information is better organized now, but this does seem like a worse search operation though :(

Reply:
I can create a std::map so it returns the ride type number by it's name.

Reply:
I think that sounds sensible? Might even be better with an `std::unordered_map`

Reply:
Best use std::find_if, this is not code that runs every frame.